### PR TITLE
Add mutually recursive definition support for AE

### PR DESCRIPTION
**Pull request by Halbaroth**
This PR adds the support for mutually recursive functions and predicates in the native language of AE.

### CODE REVIEW

**Review by Gbury — CHANGES_REQUESTED** [review]

File: src/interface/stmt.ml [hunk 1]
```diff
@@ -113,6 +113,9 @@ module type Logic = sig
   val rec_types : ?loc:location -> t list -> t
   (** Pack together a list of mutually recursive type definitions. *)
 
+  val mut_fun_def_rec : ?loc:location -> t list -> t
```
[the remark this answers is not in the document]
Comment (Halbaroth) [reply src/interface/stmt.ml:116]: Right! I though `defs` was for type definitions only.

**Comment by Halbaroth** [comment]
I added also support for mutually recursive predicates. Even if the keyword `predicate` is now a alias for function with return type `Prop`, it is nicer to write `predicate .... and predicate ...`. Notice, we cannot mix the two constructions.

**Review by Gbury — APPROVED** [review]

**Comment by Halbaroth** [comment]
@hra687261 told me `function` and `predicate` can be used at the same time in its implementation of mutually recursive definition. Thus, I have to modify this PR to have the same feature.

**Comment by Gbury** [comment]
@Halbaroth I commited and pushed to this PR the changes required to support a mix of `predicate` and `function` in the same mutually recursive group, so it's okay, you don't have to do it.

**Pull request merged**